pkg/build/builder/dockerutil.go

```diff
@@ -62,15 +81,27 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 	if glog.V(5) {
 		opts.OutputStream = os.Stderr
 	}
-	var err error
+
 	var retriableError = false
 
+	eventWatch, err := watchForFailedPushEvent(eventClient, ref.Namespace, ref.Name)
```

Comment:
But I'd rather avoid adding the carry patch here. Also docker daemon doesn't propagate error details in this case. And I'm reluctant to put the details into [error message ](https://github.com/miminar/distribution/blob/master/registry/api/errcode/register.go#L30) instead of the description. This will need some more thought.